### PR TITLE
Added AppStream metadata

### PR DESCRIPTION
**Pull request by Mailaender**
I created a Flatpak package and they use the https://freedesktop.org/software/appstream/ standard including https://hughsie.github.io/oars/ plus they also require an icon that looks nice with modern resolutions.

### CODE REVIEW

File: debian/appstream/com.djoffe.gnukem.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+<?xml version='1.0' encoding='utf-8'?>
+<component type="desktop">
+  <!--Created with jdAppdataEdit 5.1-->
+  <id>davegnukem</id>
```
Comment (hfiguiere) [inline debian/appstream/com.djoffe.gnukem.metainfo.xml:4]:
```suggestion
  <id>com.djoffe.gnukem</id>
```

**Comment by hfiguiere** [comment]
(flathub reviewer here, since this is the purpose of that patch submission)

you should add the rules to the debian pacakging so that they get installed too (not blocking flatpak, but this make life easier for everyone). Ideally the main build system should install everyting, from .desktop to icons.

**Comment by hfiguiere** [comment]
actually the main makefile does install the 32x32 and .desktop so just add this icon and the metainfo to it. This is would make the flatpak manifest simpler.

**Comment by matteobin** [comment]
Thank you for your commits, @Mailaender.
I would like to ask you a couple of questions to write a more complete
Makefile the next time. Please be patient, I'm still new to Make.

Why did you add the BIN_DIR variable, did you need it to make the
Flatpack package?

What's the difference between using the [conditional variable assignment
operator](https://www.gnu.org/software/make/manual/html_node/Conditional-Assignment.html#Conditional-Assignment) (?=) and calling make [overriding variables](https://www.gnu.org/software/make/manual/html_node/Overriding.html#Overriding)?
E.g. `make BIN_DIR=/usr/games`

My understanding is that variable overriding is more portable, because
the ?= operator is not POSIX. Besides, for the Dave Gnukem Makefile, you
would have the same result on both cases, but the ?= operator would not
work on a non-GNU Make.

**Comment by Mailaender** [comment]
Without the `?` I can't override. The hard-coded `/usr/games` seems to be Debian specific.

**Comment by matteobin** [comment]
Could please try the following test?
```
user@debian:~/tmp$ cat Makefile 
VAR1  = makeVAR1
VAR2 ?= makeVAR2

all:
	@echo VAR1 = $(VAR1)
	@echo VAR2 = $(VAR2)
user@debian:~/tmp$ VAR1=envVAR1 VAR2=envVAR2 make 
VAR1 = makeVAR1
VAR2 = envVAR2
user@debian:~/tmp$ make VAR1=cliVAR1 VAR2=cliVAR2
VAR1 = cliVAR1
VAR2 = cliVAR2
```

**Comment by davidjoffe** [comment]
Thanks everyone for the help here!!

**Pull request merged**